src/Hcal/HcalDigiProducer.cxx

```diff
@@ -95,27 +145,224 @@ void HcalDigiProducer::produce(framework::Event& event) {
   hcalDigis.setNumSamplesPerDigi(nADCs_);
   hcalDigis.setSampleOfInterestIndex(iSOI_);
 
-  std::map<unsigned int, std::vector<const ldmx::SimCalorimeterHit*>> hitsByID;
+  std::map<unsigned int, std::vector<const ldmx::SimCalorimeterHit*>> hitsByID;  //old code
+  std::map<ldmx::HcalDigiID, std::vector<double>> photonTimes;  //new code
 
   // get simulated hcal hits from Geant4 and group them by id
-  auto hcalSimHits{event.getCollection<ldmx::SimCalorimeterHit>(
-      inputCollName_, inputPassName_)};
+  auto hcalSimHits{event.getCollection<ldmx::SimCalorimeterHit>(inputCollName_, inputPassName_)};
+
+  // get map of simulated particles
+  auto particleMap{event.getMap<int, ldmx::SimParticle>("SimParticles")};
 
   for (auto const& simHit : hcalSimHits) {
     // get ID
     unsigned int hitID = simHit.getID();
 
+#define NEWCODE
+#ifdef NEWCODE
+
+    //******************//
+    //***  new code  ***//
+    //******************//
+
+    ldmx::HcalID detID(simHit.getID());
+    int section = detID.section();
+    int layer = detID.layer();
+    int strip = detID.strip();
+    ldmx::HcalDigiID posEndID(section, layer, strip, 0);
+    ldmx::HcalDigiID negEndID(section, layer, strip, 1);
+
+    //TODO: temporary implementation for getting the scintillator length
+    int length=(int)(2.0*hcalGeometry.getHalfTotalWidth(section,(layer%2==1?layer+1:layer-1)) + 0.5); //rounded to closest int to avoid mismatches due to precission of floating point numbers
+    int reflectorType=(section==ldmx::HcalID::HcalSection::BACK?0:2);
```

Comment:
@ehrlich-uva - I'm starting a little review of this code. Can you clarify what does reflector Type mean? What is the meaning of 0,1 and 2?

Reply:
reflector types:
0: the counter has SiPMs on both ends, i.e. no reflector or absorber on either side.
1: the counter has a SiPM on one side, and a reflector on the other side.
2: the counter has a SiPM on one side, and an absorber on the other side.

Reply:
I see, do counters in the SideHcal have always an absorber in the other side? how do you distinguish a reflector from an absorber in the scintillatorbar object? @EinarElen this is not something implemented in our current G4 simulation right?

Reply:
As far as I understand, all SideHcal counters have absorbers on the other side. If not, I need to change the code.

It would be nice, if the presence/location of absorbers/reflectors of a counter or module came from the hcalGeometry, so that I don't have to hard code it in HcalDigiProducer.cxx (line 177). Same is true for the counter length (line 176).

The reflector or absorber is only relevant for the photons inside the embedded fiber. So it wouldn't be part of the G4 simulation, because we don't simulate individual photons at this stage. That's why this PR implements lookup tables that were generated in a standalone G4 program (outside the LDMX code) that simulates photons.